routes/projects/files/rename.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * PUT /projects/:username/:projectName/files/:pathToFile/rename
+ * The :pathToFile parameter must be URI encoded
+ * Renames the specified file with the new name given in the message body
+ * {
+ *   name: <string> required
+ * }
+*/
+
+var renameSchema = require('../../../schemas.js').rename;
+
+module.exports = function(server) {
+  server.route({
+    method: 'PUT',
+    path: '/projects/{username}/{projectName}/files/{path}/rename',
```

Comment:
@gideonthomas I think you need to update the comment above to indicate that we only allow `filename` changes, not `pathname`.